inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -0,0 +1,101 @@
+= Releasing INCEpTION
+
+== Prerequisites
+
+When deploying to an Artifactory instance, make sure that the UKP Maven Repository is with your
+user credentials are added to your machines `settings.xml`. In our case, the repository
+`ukp-oss-releases` has to be listed with a user and password that belongs to the `trusted-user`
+group. See https://dkpro.github.io/dkpro-core/pages/setup-maven[DKPro Core - Setting up Maven repository]
+for more information on how to do this.
+
+== Release steps
+
+[arabic]
+. Create a release issue in the INCEpTION Github issue tracker
+. Merge the branch you want to release (e.g. `{revnumber}`) into the
+  master branch. This can be done right away with push permissions to
+  master or requires a pull request.
+. Checkout the branch you want to release. Create a new branch `released_branch` and check
```

Comment:
I would suggest to include the version into the branch name, e.g. `release/0.8.x-rc1`.